src/gui/AutomationPatternView.cpp

```diff
@@ -313,27 +314,35 @@ void AutomationPatternView::paintEvent( QPaintEvent * )
 		}
 
 		float *values = m_pat->valuesAfter( it.key() );
-		for( int i = it.key(); i < (it + 1).key(); i++ )
+
+		QPainterPath path;
+		QPointF origin = QPointF(x_base + it.key() * ppt / MidiTime::ticksPerTact(),0.0f);
+		path.moveTo(origin);
+		path.moveTo(QPointF(x_base + it.key() * ppt / MidiTime::ticksPerTact(),values[0]));
+		for( int i = it.key() + 1; i < (it + 1).key(); i++ )
 		{
-			float value = values[i - it.key()];
-			const float x1 = x_base + i * ppt /
-						MidiTime::ticksPerTact();
-			const float x2 = x_base + (i + 1) * ppt /
-						MidiTime::ticksPerTact();
+			const float x1 = x_base + i * ppt / MidiTime::ticksPerTact();
 			if( x1 > ( width() - TCO_BORDER_WIDTH ) ) break;
+			float value = values[i - it.key()];
+			path.lineTo(QPointF(x1,value));
 
-			if( gradient() )
-			{
-				p.fillRect( QRectF( x1, 0.0f, x2 - x1, value ), lin2grad );
-			}
-			else
-			{
-				p.fillRect( QRectF( x1, 0.0f, x2 - x1, value ), col );
-			}
+		}
+		path.lineTo(x_base + ((it + 1).key()) * ppt / MidiTime::ticksPerTact(),values[(it + 1).key() - 1 - it.key()]);
```

Comment:
` ppt / MidiTime::ticksPerTact()` appears a lot of time in the code, you could make it a seperate variable so we don't calculate this all the time.